Handler/Demo.hs

```diff
@@ -0,0 +1,37 @@
+module Handler.Demo where
+
+import Import
+
+import Control.Monad (filterM)
+import Data.Maybe (listToMaybe)
+
+getDemoR :: Handler Html
+getDemoR = do
+    root <- getAppRoot
+    siteId <- runDB $ findDemoSite root
+
+    defaultLayout $ do
+        setTitle "Carnival Demo"
+        $(widgetFile "demo")
+
+findDemoSite :: Text -> DB SiteId
+findDemoSite root = do
+    sites <- selectList [SiteBaseUrl ==. root] []
+    msite <- listToMaybe <$> filterM noMembers sites
+
+    maybe (createDemoSite root) (return . entityKey) msite
+
+  where
```

Comment:
Yeah, I think it's fine to take it out.

Do we enforce uniqueness on site URL? We probably should. If we do, it won't be possible for somebody to override the site once we've created it.